dashel/dashel-posix.cpp

```diff
@@ -1078,15 +1079,25 @@ namespace Dashel
 				ioctl(fd, TIOCMBIS, &iFlags);
 			else
 				ioctl(fd, TIOCMBIC, &iFlags);
+			
+			struct sigaction act;
+			memset(&act, 0, sizeof (act));
+			act.sa_handler = &SerialStream::handle_sighup;
+			sigaction(SIGHUP, &act, &oldsigaction);
 		}
 		
 		//! Destructor, restore old serial port state
 		virtual ~SerialStream()
 		{
 			 tcsetattr(fd, TCSANOW, &oldtio);
+			 sigaction(SIGHUP, &oldsigaction, 0);
```

Comment:
I do not think that this behaviour (setting signal in constructor and restoring in the destructor) is correct. For example, imagine the following sequence of actions:
- open serial port 1
- open serial port 2
- close serial port 1

At that point, the wrong signal handler is present.

Reply:
While I agree on principle, I'm not sure a global variable is the proper solution either.
I even considered to never reset it for the lifetime of the application, that don't seem unreasonable to me.

Reply:
I agree, let's not reset it and put a note why.

Reply:
Done.
I will revisit that in the future though